extensions/graphite-emitter/src/main/java/io/druid/emitter/graphite/SendAllGraphiteEventConverter.java

```diff
@@ -0,0 +1,122 @@
+package io.druid.emitter.graphite;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.google.common.base.Joiner;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSortedSet;
+import com.metamx.emitter.service.ServiceMetricEvent;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Emits all the events instance of {@link com.metamx.emitter.service.ServiceMetricEvent}.
+ * <p/>
+ * All the dimensions will be retained and lexicographically order using dimensions name.
+ * <p/>
+ * The metric path of the graphite event is:
+ * <namespacePrefix>.[<druid service name>].[<druid hostname>].<dimensions values ordered by dimension's name>.<metric>
+ * <p/>
+ * Note that this path will be sanitized by replacing all the `.` or `space` to `_` {@link GraphiteEmitter#sanitize(String)}
+ */
+
+@JsonTypeName("all")
+public class SendAllGraphiteEventConverter implements DruidToGraphiteEventConverter
+{
+  @JsonProperty
+  private final boolean ignoreHostname;
+
+  @JsonProperty
+  private final boolean ignoreServiceName;
+
+  @JsonProperty
+  private final String namespacePrefix;
+
+  @JsonProperty
+  public String getNamespacePrefix()
+  {
+    return namespacePrefix;
+  }
+  @JsonProperty
+  public boolean isIgnoreServiceName()
+  {
+    return ignoreServiceName;
+  }
+
+  @JsonProperty
+  public boolean isIgnoreHostname()
+  {
+    return ignoreHostname;
+  }
+
+  @JsonCreator
+  public SendAllGraphiteEventConverter(
+      @JsonProperty("namespacePrefix") String namespacePrefix,
+      @JsonProperty("ignoreHostname") Boolean ignoreHostname,
+      @JsonProperty("ignoreServiceName") Boolean ignoreServiceName
+  )
+  {
+    this.ignoreHostname = ignoreHostname == null ? false : ignoreHostname;
+    this.ignoreServiceName = ignoreServiceName == null ? false : ignoreServiceName;
+    this.namespacePrefix = Preconditions.checkNotNull(namespacePrefix, "namespace prefix can not be null");
+  }
+
+  @Override
+  public GraphiteEvent druidEventToGraphite(ServiceMetricEvent serviceMetricEvent)
+  {
+    ImmutableList.Builder metricPathBuilder = new ImmutableList.Builder<String>();
+    metricPathBuilder.add(GraphiteEmitter.sanitize(this.getNamespacePrefix()));
```

Comment:
actually namespace prefix should not be sanitized since usually user will provide a doted path. thanks